// bugbot-github-issues/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _register = require('./register');

var _register2 = _interopRequireDefault(_register);

var _token = require('./token');

var _token2 = _interopRequireDefault(_token);

var _repos = require('./repos');

var _repos2 = _interopRequireDefault(_repos);

var _issues = require('./issues');

var _issues2 = _interopRequireDefault(_issues);

var _add = require('./add');

var _add2 = _interopRequireDefault(_add);

var _user = require('./user');

var _user2 = _interopRequireDefault(_user);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var describe = '\n  github\n    describe\n    register((err, link)=>)\n    token(code, (err, token)=>)\n    user(token, (err, user)=>)\n    repos(token, (err, repos[])=>)\n    issues(token, repoID, (err, issues[])=>)\n    add({token, repoID, title}, (err, issues)=>)\n';

exports.default = {
  describe: describe,
  token: _token2.default,
  register: _register2.default,
  user: _user2.default,
  repos: _repos2.default,
  issues: _issues2.default,
  add: _add2.default
};
